### ISSUE TITLE
IndexError when cloning on Windows 10

### ISSUE THREAD

**Issue by iyedg** [anishathalye/git-remote-dropbox#82]
On Windows 10, pushing a newly created repo as per the instructions in the wiki produces no errors. However when cloning that repo back, I get the following error

```
C:\Users\Iyed                                                                                                                    
(base) λ git clone "dropbox:///git-private/testing" -b develop -v                                                                
Cloning into 'testing'...                                                                                                        
debug: fetching: /git-private/testing/refs/heads/develop                                                                         
debug: fetching: /git-private/testing/objects/8a/51430a4f49cb5458f45b57f30d4d3a53d71a0b                                          
debug: fetching: /git-private/testing/objects/35/eecda6084a7aba050a7cc9aaee3f74f6c3545c                                          
debug: fetching: /git-private/testing/objects/27/de6b78d48cb8d1505670269c029737410da9d5                                          
debug: fetching: /git-private/testing/objects/04/3d4e39c02c8fff4a58b4c4f68f63f02f90618b                                          
debug: missing part of history from 4b825dc642cb6eb9a060e54bf8d69288fbee4904                                                     
Traceback (most recent call last):                                                                                               
  File "c:\users\iyed\miniconda3\lib\runpy.py", line 193, in _run_module_as_main                                                 
    "__main__", mod_spec)                                                                                                        
  File "c:\users\iyed\miniconda3\lib\runpy.py", line 85, in _run_code                                                            
    exec(code, run_globals)                                                                                                      
  File "C:\Users\Iyed\Miniconda3\Scripts\git-remote-dropbox.exe\__main__.py", line 9, in <module>                                
  File "c:\users\iyed\miniconda3\lib\site-packages\git_remote_dropbox\__init__.py", line 812, in main                            
    helper.run()                                                                                                                 
  File "c:\users\iyed\miniconda3\lib\site-packages\git_remote_dropbox\__init__.py", line 375, in run                             
    self._do_fetch(line)                                                                                                         
  File "c:\users\iyed\miniconda3\lib\site-packages\git_remote_dropbox\__init__.py", line 422, in _do_fetch                       
    self._fetch(sha)                                                                                                             
  File "c:\users\iyed\miniconda3\lib\site-packages\git_remote_dropbox\__init__.py", line 634, in _fetch                          
    queue.extend(git_referenced_objects(sha))                                                                                    
  File "c:\users\iyed\miniconda3\lib\site-packages\git_remote_dropbox\__init__.py", line 226, in git_referenced_objects          
    return [line.split()[2] for line in lines if not line.startswith('160000 commit ')]                                          
  File "c:\users\iyed\miniconda3\lib\site-packages\git_remote_dropbox\__init__.py", line 226, in <listcomp>                      
    return [line.split()[2] for line in lines if not line.startswith('160000 commit ')]                                          
IndexError: list index out of range                                                                                              
```

I am using:
- Windows 10
- Python 3.7.1,
- dropbox==9.3.0
- git-remote-dropbox==1.0.8
- requests=2.21.0=py37_0

**Comment by anishathalye** [comment]
I installed a fresh Windows 10 VM, installed Git, Python, and `pip install git-remote-dropbox`, and I tried creating a repo, pushing it, and cloning it. It worked fine for me.

I'm using:

- Windows 10
- Python 3.7.3
- dropbox==9.3.0
- git-remote-dropbox==1.0.8
- requests==2.21.0

Is this still an issue for you? Could you provide more detailed instructions to reproduce the bug?

**Comment by martin-pr** [comment]
Getting the same problem, with a slightly different stacktrace:
```
Traceback (most recent call last):
  File "/usr/local/bin/git-remote-dropbox", line 8, in <module>
    sys.exit(main())
  File "/home/martin/.local/lib/python3.7/site-packages/git_remote_dropbox/cli/helper.py", line 23, in main
    helper.run()
  File "/home/martin/.local/lib/python3.7/site-packages/git_remote_dropbox/helper.py", line 110, in run
    self._do_fetch(line)
  File "/home/martin/.local/lib/python3.7/site-packages/git_remote_dropbox/helper.py", line 171, in _do_fetch
    self._fetch(sha)
  File "/home/martin/.local/lib/python3.7/site-packages/git_remote_dropbox/helper.py", line 386, in _fetch
    queue.extend(git.referenced_objects(sha))
  File "/home/martin/.local/lib/python3.7/site-packages/git_remote_dropbox/git.py", line 158, in referenced_objects
    return [line.split()[2] for line in lines if not line.startswith('160000 commit ')]
  File "/home/martin/.local/lib/python3.7/site-packages/git_remote_dropbox/git.py", line 158, in <listcomp>
    return [line.split()[2] for line in lines if not line.startswith('160000 commit ')]
IndexError: list index out of range
```

Can share a repo that seems to reproduce the issue.

**Comment by anishathalye** [comment]
I still can't repro the issue (Windows 10, Python 3.8.2, git-remote-dropbox 1.1.2). It would be helpful if you can share the local copy of the repo (a tarball of the directory works great), as well as the repo's directory from Dropbox (you could download it as a `.zip` and upload it here).

**Comment by martin-pr** [comment]
I am on Ubuntu 19.10, with Python 3.7.5 and git-remote-dropbox 1.1.2 installed through pip.

The repo I am having problem with is my home page, and is about 180MB of data. While there is nothing too secret about it, I'd feel better sharing it through a private message, if thats ok.

**Comment by anishathalye** [comment]
Thanks! I'll continue the conversation with you via email, and when there's an update on the bug, I'll post back here for everyone else.

**Comment by anishathalye** [comment]
A quick update: figured out the bug, it has to do with Git versions before https://github.com/git/git/commit/f06ab027efd2cff4c4319354f1ad7d5f09e853a1 (so the latest release that works fine in this situation is v2.22.0). It has to do with the empty tree being in the commit history, so it should be pretty rare. There is also no data loss as a result of this, just a crash when cloning.

As a workaround, you can upgrade your installed `git`, and it should start working again. I should have a patch for this soon once I figure out the cleanest way to do it.

**Comment by anishathalye** [comment]
Thanks @iyedg and @martin-pr for reporting the issue. It should be fixed now (in 1e1940e957fc93c5f43113ed2ccc08391bd4587e) and available on PyPI as git-remote-dropbox 1.1.3.
